docs/recipes/ui/add-routing.md

```diff
@@ -0,0 +1,203 @@
+﻿# How do I add routes to a SAFE app without splitting the app into multiple models?
+
+*Written for SAFE template version 4.2.0*
+
+When building larger apps, you probably want different pages to be accessible through different URLs. In this recipe, we show you how to add routes to different pages to an application, including adding a "page not found" page that is displayed when an unknown URL is entered.
+
+## 1. Adding the Feliz router
+
+Install Feliz.Router in the client project
+
+```bash
+dotnet paket add Feliz.Router -p Client -V 3.8
+```
+
+!!! Warning "Feliz.Router versions"
+    At the time of writing, the current version of the SAFE template (4.2.0) does not work well with the latest version of Feliz.Router (4.0).
+    To work around this, we install Feliz.Router 3.8, the latest version that works with SAFE template version 4.2.0.
+
+    If you are working with a newer version of the SAFE template, it might be worth trying to install the newest version of Feliz.Router.
+    To see the installed version of the SAFE template, run in the command line:
+    
+    ```bash
+    dotnet new --list
+    ```
+
+To include the router in the Client, add `open Feliz.Router` at the top of Index.fs
+
+## 2. Adding the URL object
+
+Add the current page to the model of the client, using a new `Page` type
+
+=== "Code"
+    ```fsharp
+    type Page =
+        | TodoList
+        | NotFound
+    
+    type Model =
+        { CurrentPage: Page
+          Todos: Todo list
+          Input: string }
+    ```
+=== "Diff"
+    ```.diff
+    + type Page =
+    +     | TodoList
+    +     | NotFound
+    +
+    - type Model = { Todos: Todo list; Input: string }
+    + type Model =
+    +    { CurrentPage: Page
+    +      Todos: Todo list
+    +      Input: string }
+    ```
+
+## 3.  Parsing URLs
+
+Create a function to parse a URL to a page, including a wildcard for unmapped pages
+
+```fsharp
+let parseUrl url = 
+    match url with
+    | ["todo"] -> Page.TodoList
+    | _ -> Page.NotFound
+```
+
+## 4. Initialization when using a URL
+
+On initialization, set the current page
+
+=== "Code"
+    ```fsharp 
+    let init () : Model * Cmd<Msg> =
+        let page = Router.currentUrl () |> parseUrl
```

Comment:
`Feliz.Router` needs to be opened before this line.

Reply:
It is mentioned in the beginning, but not with a code snippet. Made it more explicit